eeps/eep-0049.md

```diff
@@ -1399,38 +1402,30 @@ is chosen following Erlang/OTP standards:
 Since `case_clause` is functionally the closest exceptions and that it carries
 a value, we choose to replicate the same form here.
 
-The reason `else_clause` is chosen over `begin_clause` because the `else` block
+The reason `else_clause` is chosen over `maybe_clause` because the `else` block
 could arguably be used in other constructs in the future, and constraining the
 exception to the block's name itself is likely more future-proof.
 
 Backwards Compatibility
 =======================
 
-The possibility of an early exit from a `begin ... end` expression
-means that variables declared within its scope are now potentially
-unsafe to use outside of it.
+The new keyword `maybe` is introduced and might clash with existing
+use of unquoted `maybe` as an atom which in practice also includes use
+as a module or function name.
 
-This is a change of behaviour that brings `begin` in line with the
-variables bound within a `case ... end` branch, a `try/catch` clause, or
-a `receive ... end` branch. The same is true of the `else` block.
-
-This lack of safety only needs to be started at the first `MatchOrReturnExpr`
-encountered, since all variables bound before respect the same semantics as the
-existing `begin ... end` expression. If this analysis is done rather than just
-declaring all variables as unsafe wholesale, then there is no backwards
-compatibility concern to be had.
-
-The need for the `<-` operator to be used in a new context means code built
-with support for the new expressions won't be portable to older Erlang
-releases. However, if all the code is rewritten _after_ the AST and reuses
-existing core Erlang components, built BEAM artifacts should work on older
-versions fine. This is, however, not a supported use case by the OTP team.
+The plan is to introduce this new feature together with a mechanism
+that let the user activate it and other new features individually per
+module and by that the new keyword `maybe` will only be a potential
```

Comment:
Will there be another EEP for this mechanism?

Will a feature become activated-by-default after some number of OTP versions?

Reply:
> Will there be another EEP for this mechanism?

Yes, there is ongoing work on both an EEP and an implementation. We are still doing internal reviews of the EEP.

> Will a feature become activated-by-default after some number of OTP versions?

If a feature turns out to be successful, we will activate it by default in a future release (and there might be a possibility that the feature could still be possible to be turned off for legacy code).